### PR TITLE
feat(integrations): rapidapi support

### PR DESCRIPTION
**Pull request by Shubhamai**
## Describe your changes
Added [RapidAPI](http://rapidapi.com) support.

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

File: packages/shared/providers.yaml [hunk 1]
```diff
@@ -3826,6 +3826,29 @@ ramp-sandbox:
...
+        apiKey:
+            type: string
+            title: API Key
+            description: The API key for your RapidAPI account
+    connection_config:
+        host:
+            type: string
+            title: Host
+            description: The host of the API (e.g. api-football-v1.p.rapidapi.com)
+            pattern: '^[a-z0-9_-]+$'
```
Comment (khaliqgant) [inline packages/shared/providers.yaml:3890]: Would something like `'^[a-z0-9_-]+(\.[a-z0-9_-]+)*$'` be better for the pattern to match on the periods?

Comment (Shubhamai) [reply packages/shared/providers.yaml:3890]: Not sure about this, for ex: here's a sample API key from RapidAPI: `b7c156af2dmgh5c635305f3744bap168553jsp75193c8367ef`

Comment (khaliqgant) [reply packages/shared/providers.yaml:3890]: But I thought this was the pattern for the host, not the value of the API key?

File: packages/shared/providers.yaml [hunk 2]
```diff
@@ -3826,6 +3826,29 @@ ramp-sandbox:
...
+        headers:
+            x-rapidapi-key: ${apiKey}
+            x-rapidapi-host: ${connectionConfig.host}
+        base_url: https://${connectionConfig.host}
+    docs: https://docs.nango.dev/integrations/all/rapidapi
+    credentials:
+        apiKey:
+            type: string
+            title: API Key
+            description: The API key for your RapidAPI account
```
Comment (bodinsamuel) [inline packages/shared/providers.yaml:3882]: You can add pattern and example here 
```suggestion
            description: The API key for your RapidAPI account
            example: b7c156af2dmgh5c635305f3744bap168553jsp75193c8367ef
            pattern: '^[a-zA-Z0-9]+$'
```

File: packages/shared/providers.yaml [hunk 3]
```diff
@@ -3826,6 +3826,29 @@ ramp-sandbox:
...
+rapidapi:
+    display_name: RapidAPI
+    categories:
+        - dev-tools
+    auth_mode: API_KEY
+    proxy:
+        headers:
+            x-rapidapi-key: ${apiKey}
+            x-rapidapi-host: ${connectionConfig.host}
+        base_url: https://${connectionConfig.host}
```
Comment (bodinsamuel) [inline packages/shared/providers.yaml]: can you change both base_url and host to be something like `https://${connectionConfig.subdomain}.rapidapi.com`?

File: packages/shared/providers.yaml [hunk 4]
```diff
@@ -3826,6 +3826,29 @@ ramp-sandbox:
...
+    credentials:
+        apiKey:
+            type: string
+            title: API Key
+            description: The API key for your RapidAPI account
+    connection_config:
+        host:
+            type: string
+            title: Host
+            description: The host of the API (e.g. api-football-v1.p.rapidapi.com)
```
Comment (bodinsamuel) [inline packages/shared/providers.yaml]: with the other change in base_url can you add `suffix: rapidapi.com`

File: packages/shared/providers.yaml (hunk 1 above)
Comment (Shubhamai) [reply packages/shared/providers.yaml:3890]: Sorry, completely missed it. Yes you are right, I think this should work for host url.

**Review by bodinsamuel — COMMENTED** [review]
Hey, thanks for your contribution. Added some comments regarding the subdomain, let me know if you want us to takeover ☺️

**Review by bodinsamuel — APPROVED** [review]
🙏🏻

**Review by khaliqgant — APPROVED** [review]

**Pull request merged**